galaxy_ng/app/access_control/access_policy.py

```diff
@@ -146,8 +150,66 @@ def get_access_policy(cls, view):
             }
         )
 
+    def _get_public_repo_pks(self):
+        return ansible_models.AnsibleRepository.objects.filter(private=False)
+
+    def _get_private_repo_pks(self, user, perm):
+        return get_objects_for_user(
+            user,
+            perm,
+            ansible_models.AnsibleRepository.objects.filter(private=True),
+        )
+
+    def get_ansible_repository_qs(self, view, qs, repo_perm):
```

Comment:
I think we should look into using annotations for the queryset. You could annotate each queryset with an attribute called "has_view_permission" that would return true when the user has the view permission on that object. Then you can just return `Ansible.repository.filter(Q(private=False) ! Q(has_view_permission=True))`, but I think that might take some work to get right